plugins/outputs/cloudwatch/cloudwatch_test.go

```diff
@@ -59,6 +59,131 @@ func TestBuildDimensions(t *testing.T) {
 	}
 }
 
+// Test that we retain as many tags as possible if the number exceeds the max number permitted
+func TestBuildDimensionsExceedMaxNumber(t *testing.T) {
+	assert := assert.New(t)
+
+	testPoint := testutil.TestMetric(1)
+
+	for i := len(testPoint.Tags()); i <= MaxDimensions; i++ {
+		testPoint.AddTag(fmt.Sprint(i), "Value"+fmt.Sprint(i))
+	}
+
+	dimensions := BuildDimensions(testPoint.Tags(), nil)
+
+	assert.Equal(MaxDimensions, len(dimensions), "Number of dimensions should be less than MaxDimensions")
+	assert.Equal(len(testPoint.Tags())-1, len(dimensions), "Should keep as many dimensions as possible")
+
+	for _, dimension := range dimensions {
+		tagValue := testPoint.Tags()[*dimension.Name]
+		assert.Equal(tagValue, *dimension.Value, "Value should be equal")
+	}
+}
+
+// Test that global dimensions are added correctly
+func TestGlobalDimensions(t *testing.T) {
+	assert := assert.New(t)
+
+	testPoint := testutil.TestMetric(1)
+	globalDimensions := map[string]string{
+		"Environment": "test",
+		"Deployment":  "green",
+	}
+	dimensions := BuildDimensions(testPoint.Tags(), globalDimensions)
+
+	tagKeys := make([]string, len(testPoint.Tags()))
+	i := 0
+	for k := range testPoint.Tags() {
+		tagKeys[i] = k
+		i += 1
+	}
+
+	sort.Strings(tagKeys)
+
+	assert.Equal(len(testPoint.Tags())+len(globalDimensions), len(dimensions), "Number of dimensions should be equal to number of tags plus the number of global dimensions")
```

Comment:
```suggestion
	assert.Len(dimensions,len(testPoint.Tags())+len(globalDimensions), "Number of dimensions should be equal to number of tags plus the number of global dimensions")
```

Reply:
I like your suggestion for using `assert.Len()` but unfortunately in that case it would mean switching the expectation and the actual values around which might be misleading if this test fails. I'll look out for other places I can use `assert.Len()` though!